pkg/ottl/expression.go

```diff
@@ -581,3 +582,33 @@ func (p *Parser[K]) newGetterFromConverter(c converter) (Getter[K], error) {
 		keys: c.Keys,
 	}, nil
 }
+
+// TimeGetter is a Getter that must return an int64.
+type TimeGetter[K any] interface {
+	// Get retrieves an int64 value.
+	Get(ctx context.Context, tCtx K) (time.Time, error)
+}
+
+// StandardTimeGetter is a basic implementation of IntGetter
+type StandardTimeGetter[K any] struct {
+	Getter func(ctx context.Context, tCtx K) (interface{}, error)
+}
+
+// Get retrieves an int64 value.
+// If the value is not an int64 a new TypeError is returned.
+// If there is an error getting the value it will be returned.
```

Comment:
```suggestion
// Get retrieves an time.Time value.
// If the value is not an time.Time a new TypeError is returned.
// If there is an error getting the value it will be returned.
```